sdcm/scan_operation_thread.py

```diff
@@ -120,6 +125,8 @@ def execute_query(
                         | FullPartitionScanReversedOrderEvent]) -> ResultSet:
         # pylint: disable=unused-argument
         self.log.debug('Will run command %s', cmd)
+        session.cluster.default_retry_policy = ExponentialBackoffRetryPolicy(**self._exp_backoff_retry_policy_params)
```

Comment:
this is a bit weird it comes next to the code executing the query, and no the code creating the session.

I  would recommend consolidating the session creating code into something like:

```python 
@property
def cql_connection(**kwargs):
    with self.fullscan_params.db_cluster.cql_connection_patient(
                    node=self.db_node,
                    user=self.fullscan_params.user,
                    password=self.fullscan_params.user_password, **kwargs) as session:
        session.cluster.default_retry_policy = ExponentialBackoffRetryPolicy(**self._exp_backoff_retry_policy_params)
        session.default_timeout = self._request_default_timeout
        yield session
```

there way too many repetitions of applying this retry, and it should be across the board for all of the session.